R/DataSpaceMab.R

```diff
@@ -110,10 +110,43 @@ DataSpaceMab <- R6Class(
       )
 
       invisible(!"try-error" %in% tries)
+    },
+
+    #' @description
+    #' Applies LANL metadata to mabs table.
+    getLanlMetadata = function(){
+      checkList <- function(x){
+        if(any(c("list", "data.frame") %in% class(x))){
+          lapply(x, function(y){
+            if("data.frame" %in% class(y)) {
+              setDT(y)
+              checkList(y)
+            } else {
+              checkList(y)
+            }
+          })
+        }
+      }
+      pullForLanlId <- function(lanl_id){
+        url <- paste0("https://www.hiv.lanl.gov/mojo/immunology/api/v1/epitope/ab?id=", lanl_id)
+        if(is.na(lanl_id)) return(NA)
+        dat <- tryCatch({
+          page <- suppressWarnings(readLines(url))
```

Comment:
I would use `httr::GET` instead `readLines` to fetch data. `httr` provides a more stable way of accessing table in my opinion (i.e., status code, passing arguments, and better response object), and it will be consistent with how we already make API calls in DataSpaceR (i.e., Rlabkey and https://github.com/ropensci/DataSpaceR/blob/main/R/helpers.R#L154).